packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -328,6 +329,8 @@ const _BaseButton: React.ForwardRefRenderFunction<BladeElementRef, BaseButtonPro
   const [isPressed, setIsPressed] = React.useState(false);
   const isLink = Boolean(href);
   const childrenString = getStringFromReactText(children);
+  const isChildrenComponent = childrenString === '[object Object]';
```

Comment:
Ideally should be childrenString.match('[object Object]'), so that it can work with multiple react childrens too but React.isValidElement will suffice for the country selector usecase. 

```
<Button>
  <p>hello</p>
  <p>world</p>
</Button>
```